Tests/POSIXTests/ReaddirTests.swift

```diff
@@ -38,7 +38,7 @@ class ReaddirTests: XCTestCase {
         
         do {
             var s = dirent()
-            let n = sizeof(s.d_name.dynamicType)
+            let n = MemoryLayout.of(s.d_name).size
```

Comment:
I think the least damaging thing we can do here is copying our `_ofInstance` implementation into this file (maybe under a non-underscored name):

```swift
extension MemoryLayout {
  fileprivate static func ofInstance(_: @autoclosure () -> T) -> MemoryLayout<T>.Type {
    return MemoryLayout<T>.self
  }
}
```